### PR TITLE
App flow Added

### PR DESCRIPTION
**Pull request by SarveshChavan**
App flow documentation

#66 

## Changes made
appflow.md file added which shows app flow in text format.

## How does the solution address the problem
New contributors can read the appflow file and can understand the flow of the app.

### CODE REVIEW

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Fixes:
- add this to readme.md no need of a new file
- what is this:) write it properly in markdown, I can barely understand
image
- Some of the things like Help and Chnage theme could be changed mentioning a settings tab
- take help of gpt add improve the content and app flow

Resource:
[Markdown guide](https://www.markdownguide.org/cheat-sheet/)
For markdown syntax ref u can also check readme of this repo

**Comment by SarveshChavan** [comment]
Very sorry for the mistake, I'll correct the markdown format and also add the settings tab

**Comment by dk-a-dev** [comment]
@SarveshChavan le me know once u done with changes

**Comment by SarveshChavan** [comment]
please review the PR again, sorry for the previous commit

File: README.md
Comment (dk-a-dev) [inline README.md:1]: Remove that rules part,its understandable, one can understand from seeing ur format

Comment (SarveshChavan) [reply README.md:1]: Ok I'll do it

Comment (dk-a-dev) [reply README.md:1]: No i am doing it now
@SarveshChavan

Comment (dk-a-dev) [reply README.md:1]: i will push some commits just wait

Comment (SarveshChavan) [reply README.md:1]: Ok thanks 👍🏻

**Comment by SarveshChavan** [comment]
@dk-a-dev please review the PR

**Comment by dk-a-dev** [comment]
@SarveshChavan dw i am ur co-contributor in this issue, i will add some commit then i will consider merging it

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
[Mermaid JS graphing in markdown](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/)

I am planning to use this js tool
give ur suggestions

File: README.md [hunk 1]
```diff
@@ -37,3 +37,40 @@ This project is licensed under [MIT](https://github.com/IEEE-VIT/hacktoberfest-f
...
+
+## Rules In Writing App Flow
+1. Use -> for showing flow
+2. Bold the Screens and Widgets Name also mention it in () whether it is a screen or widget
+3. Use Italic and bold style when wirting buttons and also mention it in ()
+4. Whenever there is a screen write it's information on the next line
+5. Whenever there is a button write it's action on the same line
+6. Use ordered list for showcasing the options on particular screen or widget
+
+## App Work Flow
```
[the remark this answers is not in the document]
Comment (SarveshChavan) [reply README.md:45]: I think Mermaid JS is a nice option, Ill love to learn and use it. Let me know when to start the work again?

**Comment by dk-a-dev** [comment]
@SarveshChavan download one extension in vscode named
> Markdown Preview Mermaid Support
> refer [Doc Mermaid.js](https://mermaid.js.org/intro/getting-started.html)
> I will be also helping u in this and learning

**Comment by SarveshChavan** [comment]
@dk-a-dev I have added the app flow using mermaid js but it's not getting rendered here on GitHub but in vs code, it is visible. Check it once and let me know.

**Comment by SarveshChavan** [comment]
mermaid js app flow working fine, @dk-a-dev review it when you get free

**Comment by dk-a-dev** [comment]
> mermaid js app flow working fine, @dk-a-dev review it when you get free

is it renedering nicely for u now

**Comment by SarveshChavan** [comment]
> > mermaid js app flow working fine, @dk-a-dev review it when you get free
> 
> is it renedering nicely for u now

yes

**Comment by dk-a-dev** [comment]
@SarveshChavan 
image
How is this one....??

**Comment by SarveshChavan** [comment]
About us is not calling repo card, and our main objective I guess is to explain each object such as button screen and it's functioning, the diagram you made is more like flow of screen and the thing I suggest is button, widget, screen, drawer each small component.i like this design and I'll use this and remove the long description from my app flow diagram.

**Comment by dk-a-dev** [comment]
About ud will cal repo cards when we will ad contributors button there

**Comment by SarveshChavan** [comment]
Check the latest commit and also refer to figma. I have created the about us page let me know is it ok or not and also run ghe figma prototype and check the connection also. Once it is done check the latest app flow diagram and let me know if any changes needed.

**Comment by SarveshChavan** [comment]
@dk-a-dev what's the status of this PR.

**Comment by dk-a-dev** [comment]
It is still under review ur work is done I have made some changes @mintbomb27 is reviewing them and will merge it soon if it gets approved

**Comment by dk-a-dev** [comment]
To keep u in the loop: I modified ur work by removing colours.making each box of the same shape and aligning the content

**Comment by SarveshChavan** [comment]
What's the status of this PR.

**Comment by dk-a-dev** [comment]
ye i will be merging it by tonight

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
## Hackfest-Flutter
This app aims to simplify the process of finding repositories on GitHub, view the various contributors to the repos, and also allows the user to become a contributor to the desired project.

Kindly consider leaving a :star: if you like the repository and our organisation.


## Getting Started
* Fork it.
* Clone your forked repo and move inside it:\
`git clone https://github.com/<your-github-username>/<repo-name>.git && cd <repo-name>`

* Checkout to a new branch to work on an issue:\
`git checkout -b my-amazing-feature`

* Now you can open the project.
* Run `flutter pub get` in the project terminal and get the dependencies.
* Use `flutter run debug` to run the app in your local emulator.
* Once you're all done coding, it's time to open a PR :)
Run the following commands from the root of the project directory:
\
`git add .`\
`git commit -m "A short description about the feature."`\
`git push origin <my-amazing-feature>`

Open your forked repo in your browser and then raise a PR to the `master` branch of this repository!


## Contributing
To start contributing, check out [CONTRIBUTING.md](https://github.com/IEEE-VIT/hacktoberfest-flutter/blob/master/CONTRIBUTING.md). New contributors are always welcome to support this project. If you want something gentle to start with, check out issues labelled as `easy` or `good-first-issue`

### App Workflow
""""""
// add ```mermaid
flowchart TD;
    SplashScreen[<font size=5>Splash Screen]
    SplashScreen --> Authentication[<font size=5>Authentication Screen]
    Authentication --> HomePage[<font size=5>Home Page Screen]
    HomePage --> MenuButton[<font size=5> Drawer Widget Menu]
    MenuButton --> Home[<font size=5>Button: Home]
    MenuButton --> Profile[<font size=5>Button: User Profile]
    MenuButton --> SignOut[<font size=5>Button: Sign Out]
    MenuButton --> ChangeTheme[<font size=5>Button: Change Theme]
    HomePage --> SearchButton[<font size=5>Search prompt<br>Search Button]
    SearchButton --> RepoCards[<font size=5>Widget: Repo Cards]
    RepoCards --> RepoInfo[<font size=5>Repo Information]
    RepoCards --> DownArrow[<font size=5>Button: Down Arrow]
    RepoCards --> BookMark[<font size=5>Button: BookMark]
    DownArrow --> Visit[<font size=5> Button: Visit GitHub Page]
    Visit --> WebVisit[<font size=5> Opens GitHub Page]
    DownArrow --> Contributors[<font size=5>Button: Contributors List]
    Contributors --> ContributorsScreen[<font size=5>Contributors Screen]
    ContributorsScreen --> ContributorWidget[<font size=5>Widget: Contributor Info Cards]
    ContributorWidget --> ViewProfile[<font size=5>Button: View Profile]
    ViewProfile --> ProfilScreen[<font size=5>Contributor's Profile Page]
    ContributorWidget --> ContributorInfo[<font size=5>Contributor's Basic details]
    ContributorsScreen --> Floating[<font size=5>Button: Floating Action Button]
    Floating --> ContributorRequest[<font size=5>Requests to be <br>a contributor]
    HomePage --> BookMarkTab[<font size=5>BookMark Screen]
    BookMarkTab --> RepoCards
    HomePage --> AboutUsTab[<font size=5>About Us Screen]
    AboutUsTab --> IEEE-VIT-Repos[<font size=5>IEEE-VIT Repo's]
    IEEE-VIT-Repos --> RepoCards
    AboutUsTab --> Visit-IEEE-VIT[<font size=5>Button: Visit IEEE GitHub]
    AboutUsTab --> Contributors-IEEE-VIT[<font size=5>Button: Hackfest App Contributors ]
// add ```

## License
This project is licensed under [<font size=5>MIT](https://github.com/IEEE-VIT/hacktoberfest-flutter/blob/master/LICENSE)

> [NOTE!]  
> This project is not officially affiliated with DigitalOcean or Hacktoberfest in any manner.

**Comment by dk-a-dev** [comment]
@SarveshChavan just replace ur read me completely with above sent one and I will merge this PR

**Comment by dk-a-dev** [comment]
@SarveshChavan replace md asap i have to merge this one

**Comment by SarveshChavan** [comment]
share the markdown file this is a preview file unable to add in readme.md

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Add licesnse and not below app workflow refer shared markdown

**Comment by dk-a-dev** [comment]
> share the markdown file this is a preview file unable to add in readme.md
@SarveshChavan 
I was not able to share as preview, nvm
u figured it out

**Comment by SarveshChavan** [comment]
done

**Review by dk-a-dev — APPROVED** [review]
Good job merging
if it do not render,i will try to fix it but if fixing also doesn't work the PR might get dropped

**Pull request merged**